### PR TITLE
[cluster test] add reconfiguration experiment

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: testsuite/cluster-test/src/tx_emitter.rs [hunk 1]
```diff
@@ -279,23 +286,19 @@ impl TxEmitter {
...
-                    "query_sequence_numbers on {:?} for treasury compliance account failed: {}",
-                    client,
-                    e
-                )
-            })?[0];
-        Ok(AccountData {
-            address,
-            key_pair: self.mint_key_pair.clone(),
-            sequence_number,
-        })
```
Comment (snakech) [inline testsuite/cluster-test/src/tx_emitter.rs:298]: thx for refactoring loading account fns

**Review by snakech — APPROVED** [review]

**Comment by snakech** [comment]
Could you please add some description about this test and if possible paste a test output?

**Review by andll — APPROVED** [review]
This is super cool!

Curious - did you try this running test without latest fix for reconfiguration(for example by either reverting fixing commit, or by using `--validator-tag` to use different `master_xxx` tag for validator?

Would this test fail?

**Comment by zekun000** [comment]
Sample output
```
INFO 2020-09-12T02:46:52.536646Z testsuite/cluster-test/src/experiments/reconfiguration_test.rs:108 Epoch 96 committed
INFO 2020-09-12T02:47:02.798996Z testsuite/cluster-test/src/experiments/reconfiguration_test.rs:125 Epoch 97 committed
INFO 2020-09-12T02:47:12.962844Z testsuite/cluster-test/src/experiments/reconfiguration_test.rs:108 Epoch 98 committed
INFO 2020-09-12T02:47:23.195811Z testsuite/cluster-test/src/experiments/reconfiguration_test.rs:125 Epoch 99 committed
INFO 2020-09-12T02:47:33.409360Z testsuite/cluster-test/src/experiments/reconfiguration_test.rs:108 Epoch 100 committed
INFO 2020-09-12T02:47:43.518698Z testsuite/cluster-test/src/experiments/reconfiguration_test.rs:125 Epoch 101 committed
INFO 2020-09-12T02:47:53.721552Z testsuite/cluster-test/src/experiments/reconfiguration_test.rs:142 Epoch 102 committed
INFO 2020-09-12T02:47:53.721583Z testsuite/cluster-test/src/main.rs:669 Experiment finished, waiting until all affected validators recover
INFO 2020-09-12T02:47:53.721586Z testsuite/cluster-test/src/main.rs:728 Waiting for all nodes to be healthy
INFO 2020-09-12T02:47:58.752731Z testsuite/cluster-test/src/main.rs:749 All nodes are now healthy. Checking json rpc endpoints of validators and full nodes
INFO 2020-09-12T02:47:58.757976Z testsuite/cluster-test/src/main.rs:774 All json rpc endpoints are healthy
INFO 2020-09-12T02:47:58.757990Z testsuite/cluster-test/src/main.rs:677 Experiment completed
INFO 2020-09-12T02:47:58.758358Z testsuite/cluster-test/src/main.rs:605
====json-report-begin===
{
  "metrics": [],
  "text": ""
}
====json-report-end===
```

@andll which fix are you referring to? The test would fail if the txn can't be committed or timeout.

So the next step is to 
1. add some report 
2. assert the epoch number (currently the experiment succeeds if the txn aborted but committed) 
3. add to nightly suite

**Comment by andll** [comment]
>@andll which fix are you referring to? 

Correct me if I am saying some nonce, but this is what I thought - we had some breakage on premainnet related to reconfiguration transaction killing the cluster. I assume we fixed that(right?). So this cluster test experiment passes now. But, if we were to run it on libra node _before_ that fix, would this experiment fail? E.g. would it detect this failure before premainnet? Hope this makes some sense

**Comment by zekun000** [comment]
@andll are you referring to the last friday's breakage? that one is due to the vault token expiration, so the fix is also on the terraform side. I don't remember other reconfiguration related issues recently.

**Comment by andll** [comment]
@zekun000 got it. I did not realize it was vault specific issue.

**Pull request closed without merging**